### DISCUSSION TITLE
[2024-02-19] GUI Team Demo Day

### DISCUSSION THREAD

**Discussion by farmaazon** [enso-org/enso#9097] [Weekly Demo Day]
**When GUI1 will be removed?**
All known bugs are fixed. The remaining features: bringing back Code Editor (review in progress, should be merged soon) and displaying warnings (should be finished in this week)

**What's next?**
We start with enhancing drop-downs and some fixes. Next "Big thing" are file browser and buttons for execution.

![image](https://github.com/enso-org/enso/assets/3919101/e68a9101-9b81-4190-86b0-182a569e3ed2)

**Comment by MichaelMauderer** [comment]
*  #9073

[Peek 2024-02-19 12-55.webm](https://github.com/enso-org/enso/assets/1428930/9e5687b0-0365-4fec-878b-6ea3d3519c1a)
* #8678 
![Screenshot_20240219_140331](https://github.com/enso-org/enso/assets/1428930/0035e6fb-7d39-49b0-8b72-dec8d09fac7f)

**Comment by vitvakatu** [comment]
- https://github.com/enso-org/enso/pull/9093
  Although conflict detection and resolution was implemented, fully qualified names are rarely available because of engine restrictions. It means it’s likely we will need some other resolution method in the future.
- Now working on https://github.com/enso-org/enso/issues/9009

https://github.com/enso-org/enso/assets/6566674/2dab40ce-b925-4443-ab37-acd0a8968e47

**Comment by farmaazon** [comment]
- #9040 - A small change in CI tests:  they're run three times, and a single failure should block PR. I hope more transient failures will be caught.
- #9038 - the plugin was causing random failures on CI
- #9042 - fixing `Delimited_Format.Delimited` encoding parameter.

**Comment by kazcw** [comment]
<table>
  <th>
    <td>Before</td>
    <td>After</td>
    <td>Status</td>
  </th>
  <tr>
    <td>#9069 -- Draw unconnected edges on top</td>
    <td><img src="https://github.com/enso-org/enso/assets/1047859/53d1adde-8667-4e2a-b52e-5b42bc413e43"></td>
    <td><img src="https://github.com/enso-org/enso/assets/1047859/40c493db-9b91-4231-9291-7845fdc7215f"></td>
    <td>✅ Merged</td>
  </tr>
  <tr>
    <td>#9068 -- Draw target attachment</td>
    <td><img src="https://github.com/enso-org/enso/assets/1047859/b4817abf-1551-49b3-b24c-c77045c20cc8"></td>
    <td><img src="https://github.com/enso-org/enso/assets/1047859/2d31b01b-9f84-43e8-a0ec-10f3e2a0ed7a"></td>
    <td>⌛ In CI</td>
  </tr>
  <tr>
    <td>#9055 -- Code Editor</td>
    <td colspan="2"><video src="https://github.com/enso-org/enso/assets/1047859/f2929976-144f-488d-a7ba-b0f5d6e0a927"></td>
    <td>⌛ In Review</td>
  </tr>
</table>

**Comment by mwu-tow** [comment]
Mostly spent week on minor fixes and debugging.

* Fixes:
  * Fixed docker image build for the backend. #9044
  * Fixed macOS notarization regression. #9066
  * Tweaked `sbt` invocations to avoid `ServerAlreadyBootingException`. #9059
* Collecting and reporting transient failures.
* Investigating `npm install` failures on macOS GH-hosted runners.

**Comment by Frizi** [comment]
- Most of the time spent on reviewing big PRs.
- Investigated the performance issues, which turned out to be mostly related to suboptimal behavior of Vue devtools.

**Comment by hubertp** [comment]
Relevant backend work:
- #8985 - free form folder structure for projects
- #9052 - support positional arguments for text visualizations
- #9043 - synchronization issues with suggestions (reverted)
- #9077  - expression backoff support in LS
- #9057 - restart issues due to missing UUID serialization
